walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,103 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.Collections;
+import java.util.Arrays;
+import java.util.ArrayList;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    ArrayList<TimeRange> timesForMeeting = new ArrayList<TimeRange>();
+
+    // Meeting is too long  
+    if (request.getDuration() > TimeRange.WHOLE_DAY.duration()) {
```

Comment:
So, an actual issue.

This is a pretty big function. And it is very, very good programming practice to break your code up into helper functions. These will be private helpers. A good rule of thumb is don't have any function longer than ~20 lines. It's not an absolute, but you should have a hard reason to break it.

Each of these functions should be private in this case, and the name should be a overview of what the logic actually does. I will insist on this.

Reply:
Fixed